docs/developer/checkout.mdx

```diff
@@ -233,10 +548,77 @@ As a result, we get an updated checkout object with a shipping method set:
 }
 ```
 
-## Selecting a payment method
+## Payment
+
+Depending on the selected payment gateway, you will either use the JavaScript form which can be integrated to Saleor, or the payment gateway will direct you to an external payment page. The payment gateway sends information if the payment is successful, along with tokenized credit card payment information. This token is then used to run the [`checkoutPaymentCreate`](developer/api-reference/mutations/checkout-payment-create.mdx) mutation.
+
+<Chart
+  definition={`
+  sequenceDiagram
+  Client ->>+ Saleor: getPaymentMethods
+  Saleor -->>- Client: available PaymentMethods
+  Client ->>+ Saleor: checkoutPaymentCreate
+  Saleor -->>- Client: payment created
+`}
+/>
 
-Depending on the selected payment gateway, you will either use the JavaScript form which can be integrated to Saleor, or the payment gateway will direct you to an external payment page. The payment gateway sends information about if the payment is successful, along with tokenized credit card payment information. This token is then used to run the [`checkoutPaymentCreate`](developer/api-reference/mutations/checkout-payment-create.mdx) mutation.
 
+### Listing available payment gateways
+
+Available payment gateways can be listed from the `Checkout.availablePaymentGateways` field. Depending on the chosen gateway, you may need additional configuration. For example, public `api_key` is used by the frontend payment library. To obtain that data, run:
+
+```graphql
+query {
+  checkout(token: "853ca234-b113-4021-8434-b37087d652a1"){
+    availablePaymentGateways {
+      id
+      name
+      config {
+        field
+        value
+      }
+    }
+  }
+}
+```
+
+Response:
+```json
+{
+  "data": {
+    "checkout": {
+      "availablePaymentGateways": [
+        {
+          "id": "mirumee.payments.stripe",
+          "name": "Stripe",
+          "config": [
+            {
+              "field": "api_key",
+              "value": "pk_test_SkWEusCNFdSgxGGmd23z7JC1FHh3uNufNuX"
+            },
+            {
+              "field": "store_customer_card",
+              "value": "false"
+            }
+          ]
+        },
+        {
+          "id": "mirumee.payments.dummy",
+          "name": "Dummy",
+          "config": [
+            {
+              "field": "store_customer_card",
+              "value": "false"
+            }
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+### Creating payment
 The [`checkoutPaymentCreate`](developer/api-reference/mutations/checkout-payment-create.mdx) mutation requires
 
 - `token`: the checkout token
```

Comment:
This is not true. `token` is required only for PaymentGateway which demands it (for example Braintree). For Adyen, Stripe this is not required field